profiles/api.py

```diff
@@ -82,7 +82,7 @@ class Meta:
         model = Profile
 
 
-class ProfileViewSet(generics.RetrieveUpdateAPIView, viewsets.ModelViewSet):
+class ProfileViewSet(generics.RetrieveUpdateDestroyAPIView, viewsets.ModelViewSet):
```

Comment:
Hmm, instead of `ApiView` and `ModelViewSet`, I think what you want here is a combination of `Mixins` and `GenericViewSet` like in [this](http://www.django-rest-framework.org/api-guide/viewsets/#example_3) example.